plaso/parsers/esedb_plugins/ual.py

```diff
@@ -0,0 +1,587 @@
+# -*- coding: utf-8 -*-
+"""Parser for the Microsoft User Access Logging (UAL) ESE database.
+
+UAL is present in Windows Server editions starting with Server 2012.
+See: https://www.crowdstrike.com/blog/user-access-logging-ual-overview/
+"""
+
+from ipaddress import ip_address
+from uuid import UUID
+
+from dfdatetime import filetime as dfdatetime_filetime
+from dfvfs.resolver import resolver as path_spec_resolver
+from dfvfs.path import factory as path_spec_factory
+
+from plaso.containers import events
+from plaso.containers import time_events
+from plaso.lib import definitions
+from plaso.parsers import esedb
+from plaso.parsers.esedb_plugins import interface
+
+
+class UserAccessLoggingClientsEventsData(events.EventData):
+  """Windows User Access Logging CLIENTS table event data.
+
+  Attributes:
+    address (str): source IP address.
+    authenticated_username (str): domain/user account name
+        performing the access.
+    client_name (str): client name, use unknown.
+    days (List[int]): number of accesses for each day of the year.
+    role_guid (str): GUID of the service accessed.
+    role_name (str): Name of the service accessed.
+    tenant_id (str): unique GUID for a tenant client.
+    total_accesses (int): Count of accesses for the year.
+  """
+
+  DATA_TYPE = 'windows:ual:clients'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(UserAccessLoggingClientsEventsData, self).__init__(
+        data_type=self.DATA_TYPE)
+
+    self.address = None
+    self.authenticated_username = None
+    self.client_name = None
+    self.days = None
```

Comment:
number of days? days of month? what does this value represent?

Reply:
`days` was a list of the number of accesses per day for each day of the year, for a particular entry (user/ip address/role combination) in the database. It could be used to derive events (in addition to the events of first and last access) for accesses on a particular day, but those would only have precision up to the day of the access, so I didn't implement that. However, that's a possibility.